### PR TITLE
Fix 3209/improving and removing WUI error logs

### LINKED ISSUE #3209: WUI error logs
**Describe the solution you'd like**

Without even having started using the product (no agents are registered), the APP logs show error messages:

image

The Wazuh OVA also has the error logs:

image

**Describe alternatives you've considered**

Clean log.

### PR DESCRIPTION
**Pull request by CPAlejandro**
Hi guys,
In this PR we solve the problem that several records of the same type are shown when starting the app, we improve the message that is shown for the errors of the `Response Error` and `security_exception `type.

To test it, you will have to create an environment, delete the wazuh folder from / kibana / data and see the logs that appear in the app `/Configuration/Logs` section.

The tests must be done with ODFE, X-Pack and without security.

Closes #3209

### CODE REVIEW

File: server/start/monitoring/index.ts [hunk 1]
```diff
@@ -176,19 +177,16 @@ async function checkTemplate(context) {
  */
 async function insertMonitoringDataElasticsearch(context, data) {
   const monitoringIndexName = MONITORING_INDEX_PREFIX + indexDate(MONITORING_CREATION);
-  try {
     if (!MONITORING_ENABLED){
       return;
     };
-    try{
+    tryCatchForIndexPermissionError(monitoringIndexName, 'monitoring:insertMonitoringDataElasticsearch') (async() => {
```
Comment (gabiwassan) [inline server/start/monitoring/index.ts:183]: This  string`'monitoring:insertMonitoringDataElasticsearch'` could be on a `const` ?

**Review by gabiwassan — APPROVED** [review]
LGTM! 🚀

File: server/start/tryCatchForIndexPermissionError.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+import { log } from '../lib/logger';
+
+export const tryCatchForIndexPermissionError = (wazuhIndex: string, logLocation?: string, context?) => (functionToTryCatch) => async () => {
+    console.log("HOAL QUE PASA NENE", logLocation);
```
Comment (matiasmoreno876) [inline server/start/tryCatchForIndexPermissionError.ts:15]: Is this console.log correct?

File: server/start/tryCatchForIndexPermissionError.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    try {
+        await functionToTryCatch();
+    }
+    catch (error) {
+        switch(error.message){
+            case 'security_exception':
+              error.message = (((((error.meta || error.message).body || error.message).error || error.message).root_cause[0] || error.message).reason || error.message);
+              break;
+            case 'Response Error':
```
Comment (matiasmoreno876) [inline server/start/tryCatchForIndexPermissionError.ts:24]: I would pass this string 'Response Error' to constants.

File: server/start/monitoring/index.ts (hunk 1 above)
Comment (CPAlejandro) [reply server/start/monitoring/index.ts:183]: Amazing idea, thanks you!

File: server/start/tryCatchForIndexPermissionError.ts (hunk 2 above)
Comment (CPAlejandro) [reply server/start/tryCatchForIndexPermissionError.ts:15]: Nice catch, thanks you!

File: server/start/tryCatchForIndexPermissionError.ts (hunk 3 above)
Comment (CPAlejandro) [reply server/start/tryCatchForIndexPermissionError.ts]: I have changed it, thank you!

**Review by pablomarga — APPROVED** [review]
Good job! LGTM

File: server/controllers/wazuh-api.ts [hunk 4]
```diff
@@ -81,7 +81,7 @@ export class WazuhApiCtrl {
       });
     } catch (error) {
       const errorMessage = ((error.response || {}).data || {}).detail || error.message || error;
-      log('wazuh-api:getToken', errorMessage);
+        log('wazuh-api:getToken', errorMessage);
```
Comment (Desvelao) [inline server/controllers/wazuh-api.ts:84]: Remove the identation

File: server/start/cron-scheduler/save-document.ts [hunk 5]
```diff
@@ -43,7 +44,7 @@ export class SaveDocument {
 
   private async checkIndexAndCreateIfNotExists(index, shards, replicas) {
     try {
-      try {
+      tryCatchForIndexPermissionError(index, this.logPath) (async() => {
```
Comment (Desvelao) [inline server/start/cron-scheduler/save-document.ts:47]: This needs an `await`.

File: server/start/index.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
 export * from './cron-scheduler';
 export * from './initialize';
 export * from './monitoring';
-export * from './queue';
\ No newline at end of file
+export * from './queue';
+export * from './tryCatchForIndexPermissionError'
```
Comment (Desvelao) [inline server/start/index.ts:5]: Semicolon `;`

File: server/start/initialize/index.ts [hunk 7]
```diff
@@ -18,6 +18,7 @@ import fs from 'fs';
 import { ManageHosts } from '../../lib/manage-hosts';
 import { WAZUH_ALERTS_PATTERN, WAZUH_DATA_CONFIG_REGISTRY_PATH, WAZUH_INDEX, WAZUH_VERSION_INDEX, WAZUH_KIBANA_TEMPLATE_NAME, WAZUH_DATA_KIBANA_BASE_ABSOLUTE_PATH } from '../../../common/constants';
 import { createDataDirectoryIfNotExists } from '../../lib/filesystem';
+import { tryCatchForIndexPermissionError } from '../tryCatchForIndexPermissionError'
```
Comment (Desvelao) [inline server/start/initialize/index.ts:21]: Semicolon `;`

File: server/start/initialize/index.ts [hunk 8]
```diff
@@ -365,7 +357,8 @@ export function jobInitializeRun(context) {
         await getTemplateByName();
       }
     } catch (error) {
-      log('initialize:checkKibanaStatus', error.message || error);
+      if(error.name !== 'ResponseError')
+        log('initialize:checkKibanaStatus', error.message || error);
       context.wazuh.logger.error(error.message || error);
```
Comment (Desvelao) [inline server/start/initialize/index.ts:362]: The condition is only affecting the `log` function and no to Kibana logger. Should it be of this way?

File: server/start/monitoring/index.ts [hunk 9]
```diff
@@ -26,6 +26,7 @@ import {
   WAZUH_MONITORING_DEFAULT_ENABLED,
   WAZUH_MONITORING_DEFAULT_FREQUENCY,
 } from '../../../common/constants';
+import { tryCatchForIndexPermissionError } from '../tryCatchForIndexPermissionError'
```
Comment (Desvelao) [inline server/start/monitoring/index.ts:29]: Semicolon `;`

File: server/start/tryCatchForIndexPermissionError.ts [hunk 10]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Wazuh app - React HOCs to manage the message when elastic show a Response error
```
Comment (Desvelao) [inline server/start/tryCatchForIndexPermissionError.ts:2]: It is not a `HOC`, it is a `HOF` (High order function)

File: server/start/tryCatchForIndexPermissionError.ts [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+ * Find more information about this on the LICENSE file.
+ */
+import { log } from '../lib/logger';
+
+export const tryCatchForIndexPermissionError = (wazuhIndex: string, logLocation?: string, context?) => (functionToTryCatch) => async () => {
+    try {
+        await functionToTryCatch();
+    }
+    catch (error) {
+        enum error_types{
```
Comment (Desvelao) [inline server/start/tryCatchForIndexPermissionError.ts:19]: Maybe you could name in uppercase

**Review by Desvelao — CHANGES_REQUESTED** [review]
Review the suggetions

**Comment by Desvelao** [comment]
Testing ✅ 
image

File: server/controllers/wazuh-api.ts (hunk 4 above)
Comment (CPAlejandro) [reply server/controllers/wazuh-api.ts:84]: Done!

File: server/start/index.ts (hunk 6 above)
Comment (CPAlejandro) [reply server/start/index.ts:5]: Thanks you!

File: server/start/tryCatchForIndexPermissionError.ts (hunk 10 above)
Comment (CPAlejandro) [reply server/start/tryCatchForIndexPermissionError.ts:2]: Thta's true, thanks you!

File: server/start/tryCatchForIndexPermissionError.ts (hunk 11 above)
Comment (CPAlejandro) [reply server/start/tryCatchForIndexPermissionError.ts:19]: Done!

File: server/start/monitoring/index.ts (hunk 9 above)
Comment (CPAlejandro) [reply server/start/monitoring/index.ts:29]: Thanks you!

File: server/start/initialize/index.ts (hunk 7 above)
Comment (CPAlejandro) [reply server/start/initialize/index.ts:21]: Thanks you!

File: server/start/initialize/index.ts (hunk 8 above)
Comment (CPAlejandro) [reply server/start/initialize/index.ts:362]: Thanks, if it could be interesting to try to control the kibana loggers

**Review by pablomarga — APPROVED** [review]
Tested X-Pack: :heavy_check_mark: 
Tested ODFE: :heavy_check_mark: 
Tested no security: :heavy_check_mark:

**Review by Desvelao — APPROVED** [review]
LGTM!
CR: ✅ 
Testing: ✅

**Pull request merged**